apps/designer/app/designer/features/sidebar-left/panels/pages/pages.tsx

```diff
@@ -0,0 +1,126 @@
+import { TreeNodeLabel, TreeNode } from "@webstudio-is/design-system";
+import { type Publish } from "~/shared/pubsub";
+import { PageIcon } from "@webstudio-is/icons";
+import type { TabName } from "../../types";
+import { Header } from "../../lib/header";
+import { type Page, type Pages, type Project } from "@webstudio-is/project";
+import { useMemo } from "react";
+import { useNavigate } from "react-router-dom";
+import { type Config } from "~/config";
+import {
+  useCurrentPageId,
+  usePages,
+  useProject,
+} from "~/designer/shared/nano-states";
+
+type TabContentProps = {
+  onSetActiveTab: (tabName: TabName) => void;
+  publish: Publish;
+  config: Config;
+};
+
+type PagesTreeNode =
+  | {
+      // currently used only for root node
+      type: "folder";
+      id: string;
+      children: PagesTreeNode[];
+    }
+  | {
+      type: "page";
+      id: string;
+      data: Page;
+    };
+
+const toTreeData = (pages: Pages): PagesTreeNode => {
+  return {
+    type: "folder",
+    id: "root",
+    children: [pages.homePage, ...pages.pages].map((data) => ({
+      type: "page",
+      id: data.id,
+      data,
+    })),
+  };
+};
+
+const staticTreeProps = {
+  getItemChildren(node: PagesTreeNode) {
+    if (node.type === "folder") {
+      return node.children;
+    }
+    return [];
+  },
+  getIsExpanded(_node: PagesTreeNode) {
+    return true;
+  },
+  renderItem(props: { data: PagesTreeNode; isSelected: boolean }) {
+    if (props.data.type === "folder") {
+      return null;
+    }
+
+    return (
+      <TreeNodeLabel
+        isSelected={props.isSelected}
+        text={props.data.data.name}
+      />
+    );
+  },
+};
+
+export const TabContentWithData = ({
+  onSetActiveTab,
```

Comment:
pass onClose, move onSetActiveTab to TabContent, so that Pages has nothing to do with Tabs

Reply:
I did a similar thing for image manager recently, because we started reusing image manager and fonts manager UI that we use in tabs also in a floating window, we will need the same for pages because we will want to be able to link a page from props panel, so that page manager acts like a page picker

Reply:
Oh btw you willl want to use the hooks inside the Pages component, because you will want that pages component to get the data automatically when used as a picker in floating window